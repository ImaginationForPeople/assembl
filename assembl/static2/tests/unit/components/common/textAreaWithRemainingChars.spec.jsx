import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import TestedComponent from '../../../../js/app/components/common/textAreaWithRemainingChars';
import '../../../helpers/setupTranslations';

describe('TextAreaWithRemainingChars component', () => {
  const onChangeSpy = jest.fn(() => {});
  const onClickSpy = jest.fn(() => {});
  it('should render a textarea field with a limited number of characters (empty value)', () => {
    const props = {
      domId: 'my-textarea',
      maxLength: 600,
      placeholder: 'Use the mobile ADP capacitor, then you can calculate the mobile port!',
      rows: 4,
      onChange: onChangeSpy,
      onClick: onClickSpy,
      value: ''
    };
    const renderer = new ShallowRenderer();
    renderer.render(<TestedComponent {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });

  it('should render a textarea field with a limited number of characters', () => {
    const props = {
      domId: 'my-textarea',
      maxLength: 600,
      placeholder: 'Use the mobile ADP capacitor, then you can calculate the mobile port!',
      rows: 4,
      onChange: onChangeSpy,
      onClick: onClickSpy,
      value: 'Try to bypass the USB port, maybe it will generate the wireless sensor!'
    };
    const renderer = new ShallowRenderer();
    renderer.render(<TestedComponent {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});