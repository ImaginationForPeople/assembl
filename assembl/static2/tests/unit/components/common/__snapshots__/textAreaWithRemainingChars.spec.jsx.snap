// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaWithRemainingChars component should render a textarea field with a limited number of characters (empty value) 1`] = `
<div>
  <FormControl
    bsClass="form-control"
    className="txt-area"
    componentClass="textarea"
    id="my-textarea"
    maxLength={600}
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Use the mobile ADP capacitor, then you can calculate the mobile port!"
    rows={4}
    value=""
  />
  <div
    className="annotation margin-xs"
  >
    <Translate
      nbCharacters={600}
      value="debate.remaining_x_characters"
    />
  </div>
</div>
`;

exports[`TextAreaWithRemainingChars component should render a textarea field with a limited number of characters 1`] = `
<div>
  <FormControl
    bsClass="form-control"
    className="txt-area"
    componentClass="textarea"
    id="my-textarea"
    maxLength={600}
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Use the mobile ADP capacitor, then you can calculate the mobile port!"
    rows={4}
    value="Try to bypass the USB port, maybe it will generate the wireless sensor!"
  />
  <div
    className="annotation margin-xs"
  >
    <Translate
      nbCharacters={529}
      value="debate.remaining_x_characters"
    />
  </div>
</div>
`;
